src/cljs/pyregence/components/settings/nav_bar.cljs

```diff
@@ -0,0 +1,200 @@
+(ns pyregence.components.settings.nav-bar
+  (:require
+   [clojure.string                       :as str]
+   [clojure.walk                         :as walk]
+   [herb.core                            :refer [<class]]
+   [pyregence.components.svg-icons       :as svg]
+   [reagent.core                         :as r]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; CSS Styles
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn- $on-hover-darken-background []
+  (with-meta {} {:pseudo {:hover {:background "#F8E8CC"}}}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; UI Components
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn- button
+  [{:keys [icon text on-click selected?]}]
+  [:nav-button {:on-click on-click
+                :class    (<class $on-hover-darken-background)
+                :style    (cond-> {:display "flex"
+                                   :height "52px"
+                                   :padding "16px"
+                                   :gap "12px"
+                                   :align-items "center"
+                                   :align-self "stretch"}
+                            selected?
+                            (assoc
+                             :background "#F8E8CC"
+                             :border-left "5px solid #E58154"))}
+   (when icon [icon :height "24px" :width "24px"])
+   [:label {:style {:color       "#4A4A4A"
+                    :text-align  "justify"
+                    :font-family "Roboto"
+                    :font-size   "16px"
+                    :font-style  "normal"
+                    :font-weight "400"
+                    :line-height "16px"}} text]])
+
+(defn- same-letters-so-far? [s1 s2]
+  (let [s1 (clojure.string/lower-case s1)
```

Comment:
Would be better to :require clojure.string :as something IMO

Reply:
You're already requiring `clojure.string` in this namespace, so change `clojure.string/lower-case` to just `str/lower-case`.

```suggestion
  (let [s1 (str/lower-case s1)
```